### PR TITLE
feat(examples): Vertex  Machine Learning Pipeline

### PR DESCRIPTION
**Pull request by renato-rudnicki**
This PR creates a Machine Learning Pipeline example with the following:

- Refactor the previous machine learning pipeline example
- Fixes cloud build service account breaking issues
- Adds service accounts / permissions / vpc-sc rules necessary to run the example
- Fix the documentation

### CODE REVIEW

**Review by caetano-colin — APPROVED** [review]
Tested in pair with Renato - LGTM

File: 2-environments/README.md [hunk 1]
```diff
@@ -196,7 +196,7 @@ You will be doing this procedure for each environment (`development`, `non-produ
     Make sure your git is checked out to the development branch by running `git checkout development` on `GCP_ENVIRONMENTS_PATH`.
 
     ```bash
-    (cd $GCP_ENVIRONMENTS_PATH && git checkout development)
+    (cd $GCP_ENVIRONMENTS_PATH && git checkout development && cd $GCP_ENVIRONMENTS_PATH/envs/development && terraform init)
```
Comment (caetano-colin) [inline 2-environments/README.md:199]:
```suggestion
    (cd $GCP_ENVIRONMENTS_PATH && git checkout development && ./tf-wrapper.sh init development)
```

File: 2-environments/README.md [hunk 2]
```diff
@@ -244,7 +244,7 @@ You will be doing this procedure for each environment (`development`, `non-produ
     Make sure your git is checked out to the `non-production` branch by running `git checkout non-production` on `GCP_ENVIRONMENTS_PATH`.
 
     ```bash
-    (cd $GCP_ENVIRONMENTS_PATH && git checkout non-production)
+    (cd $GCP_ENVIRONMENTS_PATH && git checkout non-production && cd $GCP_ENVIRONMENTS_PATH/envs/non-production && terraform init)
```
Comment (caetano-colin) [inline 2-environments/README.md:247]:
```suggestion
    (cd $GCP_ENVIRONMENTS_PATH && git checkout non-production && ./tf-wrapper.sh init non-production)
```

File: 2-environments/README.md [hunk 3]
```diff
@@ -292,7 +292,7 @@ You will be doing this procedure for each environment (`development`, `non-produ
     Make sure your git is checked out to the `production` branch by running `git checkout production` on `GCP_ENVIRONMENTS_PATH`.
 
     ```bash
-    (cd $GCP_ENVIRONMENTS_PATH && git checkout production)
+    (cd $GCP_ENVIRONMENTS_PATH && git checkout production && cd $GCP_ENVIRONMENTS_PATH/envs/production && terraform init)
```
Comment (caetano-colin) [inline 2-environments/README.md:295]:
```suggestion
    (cd $GCP_ENVIRONMENTS_PATH && git checkout production && ./tf-wrapper.sh init production)
```

**Review by caetano-colin — APPROVED** [review]
Made some suggestions, overall LGTM

File: scripts/validate-requirements.sh [hunk 4]
```diff
@@ -305,9 +321,11 @@ function main(){
 
     echo "Validating 0-bootstrap configuration..."
     if [[ "$EXTERNAL_REPO" == "true" ]]; then
-        validate_bootstrap_step_external_repo
+    	validate_bootstrap_step_external_repo
+    elif [[ "$TERRAFORM_LOCAL" == "true" ]]; then
+    	validate_bootstrap_terraform_local
```
Comment (apeabody) [inline scripts/validate-requirements.sh:326]:
```suggestion
    	validate_bootstrap_terraform_local
```

File: examples/machine-learning-pipeline/README.md [hunk 5]
```diff
@@ -1,51 +1,28 @@
...
-</tr>
-</tbody>
-</table>
-
-For an overview of the architecture and the parts, see the
-[terraform-google-enterprise-genai README](https://github.com/GoogleCloudPlatform/terraform-google-enterprise-genai)
-file.
+This example demonstrates the process of interactive coding and experimentation using the Google Vertex AI Workbench for data scientists. The guide outlines the creation of a machine learning (ML) pipeline within a notebook on a Google Vertex AI Workbench Instance.
+
+This environment is set up for interactive coding and experimentations. After the project is up, the vertex workbench is deployed from service catalog and the datascientis can use it to write their code including any experiments, data processing code and pipeline components. In addition, a cloud storage bucket is deployed to use as the storage for our operations. Optionally a composer environment is which will later be used to schedule the pipeline run on intervals.
```
Comment (caetano-colin) [inline examples/machine-learning-pipeline/README.md:5]:
```suggestion
This environment is set up for interactive coding and experimentations. After the project is up, the vertex workbench will be deployed from the base environment module on (`/modules/base_env/main.tf` and the data scientists can use it to write their data processing code and pipeline components. In addition, a cloud storage bucket should be deployed to use as the storage for our operations. Optionally, a composer environment can be setup to schedule the pipeline run on intervals.
```

File: examples/machine-learning-pipeline/README.md [hunk 6]
```diff
@@ -1,51 +1,28 @@
...
+  - Use a notebook on Google Vertex AI Workbench Instance to develop and adjust the ML pipeline on the development environment.
+- Triggering the Pipeline:
+  - The pipeline is set to trigger via Cloud Build upon merges to the non-production branch after being validated on development environment.
+- Training and Deploying the Model:
+  - The model is trained and deployed using the census income dataset.
+  - Deployment and monitoring occur in the production environment.
+- A/B Testing:
+  - After successful pipeline runs, a new model version is deployed for A/B testing.
 
 ## Purpose
```
Comment (caetano-colin) [inline examples/machine-learning-pipeline/README.md:27]: remove purpose section or correct the text

File: examples/machine-learning-pipeline/README.md [hunk 7]
```diff
@@ -420,519 +672,1534 @@ Run `terraform output cloudbuild_project_id` in the `0-bootstrap` folder to get
    cd ..
    ```
 
-## Running Terraform locally
+### VPC-SC - Infrastructure Deployment with Local Terraform - Only proceed with these if you have not used Cloud Build
```
Comment (caetano-colin) [inline examples/machine-learning-pipeline/README.md:675]: Move the section `VPC-SC - Infrastructure Deployment with Local Terraform - Only proceed with these if you have not used Cloud Build` as a subsection to `Prerequisites` after the VPC-SC instructions for cloudbuild

Comment (caetano-colin) [reply examples/machine-learning-pipeline/README.md:675]: and remove the duplicated "Usage" sections that is right above it

**Review by caetano-colin — CHANGES_REQUESTED** [review]
overall LGTM - only adding some changes on the documentation

File: examples/machine-learning-pipeline/README.md (hunk 7 above)
Comment (renato-rudnicki) [reply examples/machine-learning-pipeline/README.md:698]: I created [this issue ](https://github.com/GoogleCloudPlatform/terraform-google-enterprise-genai/issues/68)as an improvement. I believe we having two separated READMEs will keep the doc more organized.

**Pull request merged**